dask/dataframe/tests/test_shuffle.py

```diff
@@ -611,7 +611,7 @@ def test_set_index_interpolate():
 
     d1 = d.set_index("x", npartitions=3)
     assert d1.npartitions == 3
-    assert set(d1.divisions) == set([1, 2, 3, 4])
+    assert set(d1.divisions) == set([1, 2, 4])
```

Comment:
@eriknw do you think this is an issue?

Reply:
Nope.  This test is contrived and the behavior doesn't reflect a real concern.

I think it would be very hard to contrive a pathological example where this change would make a noticeable impact (and if I'm wrong about this, then I would be very interested in learning where it does make an impact!).  The likely worst case scenario is other projects will need to update a couple tests that depended on the old behavior (but even this may be a stretch).

Everything LGTM.

I currently don't have write permission to this repo, so somebody else will need to merge.